FTPFolderUploader/main.py

```diff
@@ -0,0 +1,293 @@
+                        continue_on = True
+                    except Exception as e:
+                        print('ERROR!')
+                        print(str(e.args))
+                        print()
+                else:
+                    print("WARNING -- File no longer exists, (%s)!" % source)
+
+            connection.close()
+            print('Closing Connection')
+    else:
+        print('ERROR -- No files found in (%s) walkdir=%s' % (local_dir, walkdir))
+
+    return continue_on
+
+
+def mkdir_p(sftp, remote_directory):
+
+    dir_path = str()
+    new_dir = False
+
+    for dir_folder in remote_directory.split("/"):
+        if dir_folder == "":
+            continue
+        dir_path += r"/{0}".format(dir_folder)
+        curr_path = dir_path.strip("/")
+        try:
+            sftp.listdir(curr_path)
+        except IOError:
+            sftp.mkdir(curr_path)
+            new_dir = True
+
+    if new_dir:
+            print("Created %s" % remote_directory)
+
+
+if __name__ == '__main__':
+    import optparse
+
+    default_config_file = u'ftp.conf'
+
+    # Create parser, and configure command line options to parse
+    parser = optparse.OptionParser()
+    parser.add_option("-l", "--local_dir",
+                      dest="local_dir",
+                      help="[REQUIRED] Local Directory",
+                      default=None)
+    parser.add_option("-u", "--username",
+                      dest="username",
+                      help="[REQUIRED] username",
+                      default=None)
+    parser.add_option("-s","--server",
+                      dest="server",
+                      help="[REQUIRED] Server Address",
+                      default=None)
+    parser.add_option("-m",
+                      action="store_true",
+                      dest="monitor",
+                      help="Monitor directory changes SLEEP_SECONDS: %s" % SLEEP_SECONDS,
+                      default=None)
+    parser.add_option("-w",
+                      action="store_true",
+                      dest="walkdir",
+                      help="Walk sub directories of the given directory to find files to send.",
+                      default=None)
+
+    (options,args) = parser.parse_args()
+
+    if (options.username and options.server) or os.path.exists(default_config_file):
+
+        if os.path.exists(default_config_file):
+            sys.path.append('.')
+
+            config = configparser.ConfigParser()
+            config.read(default_config_file)
+
+            try:
+                server = config['ftp_conf']['server'] if options.server is None else options.server
+                username = config['ftp_conf']['username'] if options.username is None else options.username
+                local_dir = config['ftp_conf']['local_dir'] if options.local_dir is None else options.local_dir
+                monitor = eval(config['ftp_conf']['monitor']) if options.monitor is None else options.monitor
```

Comment:
This has can be cast to a bool (assuming that is the intended behaviour here?) using https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean instead of eval